uiowa/uiowa
# Classrooms migration fields update

Resolves #6179 





# How to test


One line option to take you up through opening the migrate config page
`git checkout classrooms-migrate-update && git pull && ddev blt ds --site=classrooms.uiowa.edu && ddev drush @classrooms.local en classrooms_migrate -y && ddev drush @classrooms.local uli /admin/config/sitenow/sitenow-migrate`

Sync the site in devdekstop on uiowa703
Pull the branch
Sync classrooms.uiowa.edu
enable the `classrooms_migrate` module
Log into the site and visit /admin/config/sitenow/sitenow-migrate
Complete the form according to the help text, matching the data in devdesktop and setting the "Base URL" path to https://classrooms.uiowa.edu

`ddev drush @classrooms.local ms` to check that the migration registered properly. Should see over 300 unprocessed waiting
`ddev drush @classrooms.local mim classrooms_room`
Check the migrated rooms, and compare with the source room pages.

Check that the URL alias pattern matches that of the existing site (/buildingAbbreviation-roomNumber, such as /abw-110)

Basic description of the mappings:

- Building -> Building - I believe the D9 select list has a key of building code so hopefully we don't need some lookup or something tomigrate building code to building name
- Room ID -> Number
- Image Gallery -> Images - multiple
- Accessibility -> Don't need to map because this field didn't previously exist
- Tile Details -> Tile Details - combine name and details field collection items into string - multiple values
- Design Details -> Design Details - combine name and detail field collection items into string - multiple values
- Furniture -> Classroom Furniture - combine type and details field collection items into string - multiple values
- Body -> Don't need to map because this field didn't previously exist
- Technology Guide -> Link
- Seating Chart -> Seating Chart
- Video -> Video
- Map -> lets not do anything with this

# Deployment

We need to either remove all Room nodes in prod, or at least ensure that there is no info in the "Seating Chart" field for deployment to succeed. For further coverage, should probably remove past revisions for Room content type as well.

Alternatively, no worries if all Room content can be removed prior to release.
